Makefile

```
@@ -68,6 +68,29 @@ checkgenerate:
 	@# Used in CI to verify that `make generate` doesn't produce a diff.
 	test -z "$$(git status --porcelain | tee /dev/stderr)"
 
+.PHONY: runservertests
+runservertests: $(BIN)/connectconformance $(BIN)/referenceserver
+	$(BIN)/connectconformance --mode server $(BIN)/referenceserver
+
+.PHONY: runclienttests
+runclienttests: $(BIN)/connectconformance $(BIN)/referenceclient
+	$(BIN)/connectconformance --mode client $(BIN)/referenceclient
+
+$(BIN)/connectconformance: Makefile generate
+	go build -o $(@) cmd/connectconformance/main.go
+
+$(BIN)/referenceclient: Makefile generate
+	go build -o $(@) cmd/referenceclient/main.go
+
+$(BIN)/referenceserver: Makefile generate
+	go build -o $(@) cmd/referenceserver/main.go
+
+$(BIN)/grpcclient: Makefile generate
+	go build -o $(@) cmd/grpcclient/main.go
+
+$(BIN)/grpcserver: Makefile generate
+	go build -o $(@) cmd/grpcserver/main.go
```

It's best to not build a single file. If we want to add code to the `main` package in a different file, this will prevent it from successfully compiling. Instead, just build the whole package.
```suggestion
$(BIN)/connectconformance: Makefile generate
	go build -o $(@) cmd/connectconformance/

$(BIN)/referenceclient: Makefile generate
	go build -o $(@) cmd/referenceclient/

$(BIN)/referenceserver: Makefile generate
	go build -o $(@) cmd/referenceserver/

$(BIN)/grpcclient: Makefile generate
	go build -o $(@) cmd/grpcclient/

$(BIN)/grpcserver: Makefile generate
	go build -o $(@) cmd/grpcserver/
```